kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiBatchService.scala

```diff
@@ -83,26 +84,28 @@ class KyuubiBatchService(
               metadata.requestArgs,
               Some(metadata),
               fromRecovery = false)
-            val sessionHandle = sessionManager.openBatchSession(batchSession)
+            sessionManager.openBatchSession(batchSession)
             var submitted = false
             while (!submitted) { // block until batch job submitted
-              submitted = sessionManager.getBatchSession(sessionHandle).map { batchSession =>
-                val batchState = batchSession.batchJobSubmissionOp.getStatus.state
-                batchState == OperationState.RUNNING || OperationState.isTerminal(batchState)
-              }.getOrElse {
-                error(s"Batch Session $batchId is not existed, marked as finished")
-                true
+              submitted = metadataManager.getBatchSessionMetadata(batchId) match {
+                case Some(metadata) if OperationState.isTerminal(metadata.opState) =>
+                  true
+                case Some(metadata) if metadata.opState == OperationState.RUNNING =>
+                  metadata.appState match {
+                    // app that is not submitted to resource manager
```

Comment:
Maybe we can assume when submit operation is running, 
- app state in PENDING, RUNNING, FINISHED, KILLED, FAILED means been submitted(return true)
- app state in NOT_FOUND or UNKNOWN we should marked as un-submitted and wait for out-range fail.

Reply:
The purpose of such concurrency limitation is that the SparkSubmit process swallows too many CPU and memory resources, and the SparkSubmit process won't exit when the app state is PENDING, thus PENDING should be treated as not submitted